src/config.rs

```diff
@@ -360,6 +367,10 @@ pub struct Pool {
 
     pub shards: BTreeMap<String, Shard>,
     pub users: BTreeMap<String, User>,
+
+    pub auth_query: Option<String>,
+    pub auth_query_user: Option<String>,
+    pub auth_query_password: Option<String>,
```

Comment:
Just noticed this: we need to move these fields up above the `BTreeMap` fields (see comment below).